gremlin-go/driver/graphBinary.go

```diff
@@ -199,6 +201,92 @@ func mapReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return valMap, nil
 }
 
+// Golang stores BigIntegers with big.Int types
+// it contains an unsigned representation of the number and uses a boolean to track +ve and -ve
+// getSignedBytesFromBigInt gives us the signed(two's complement) byte array that represents the unsigned byte array in
+// big.Int
+func getSignedBytesFromBigInt(n *big.Int) []byte {
+	var one = big.NewInt(1)
+	if n.Sign() == 0 {
+		return []byte{}
+	}
+	if n.Sign() == 1 {
+		// add a buffer 0x00 byte to the start of byte array if number is positive and has a 1 in its MSB
+		b := n.Bytes()
+		if b[0]&0x80 > 0 {
+			b = append([]byte{0}, b...)
+		}
+		return b
+	}
+	// Convert Unsigned byte array to signed byte array
+	length := uint(n.BitLen()/8+1) * 8
+	b := new(big.Int).Add(n, new(big.Int).Lsh(one, length)).Bytes()
+	// Strip any redundant 0xff bytes from the front of the byte array if the following byte starts with a 1
+	if len(b) >= 2 && b[0] == 0xff && b[1]&0x80 != 0 {
+		b = b[1:]
+	}
+	return b
+}
+
+func getBigIntFromSignedBytes(b []byte) *big.Int {
+	var newBigInt = big.NewInt(0).SetBytes(b)
+	var one = big.NewInt(1)
+	if len(b) == 0 {
+		return newBigInt
+	}
+	// If the first bit in the first element of the byte array is a 1, we need to interpret the byte array as a two's complement representation
+	if b[0]&0x80 == 0x00 {
+		newBigInt.SetBytes(b)
+		return newBigInt
+	}
+	// Undo two's complement to byte array and set negative boolean to true
+	length := uint((len(b)*8)/8+1) * 8
+	b2 := new(big.Int).Sub(newBigInt, new(big.Int).Lsh(one, length)).Bytes()
+	// strip the resulting 0xff byte at the start of array
+	b2 = b2[1:]
+	// strip any redundant 0x00 byte at the start of array
+	if b2[0] == 0x00 {
```

Comment:
Nit grammar updates and some minor spacing:
```suggestion
	// Undo two's complement to byte array and set negative boolean to true
	length := uint((len(b)*8)/8+1) * 8
	b2 := new(big.Int).Sub(newBigInt, new(big.Int).Lsh(one, length)).Bytes()
	
	// Strip the resulting 0xff byte at the start of array
	b2 = b2[1:]
	
	// Strip any redundant 0x00 byte at the start of array
	if b2[0] == 0x00 {
```